src/toil/jobStores/aws/jobStore.py

```diff
@@ -307,7 +307,8 @@ def _extractKeyInfoFromUrl(url):
         """
         s3 = boto.connect_s3()
         bucket = s3.get_bucket(url.netloc)
-        key = bucket.new_key(url.path[1:])
+        key = bucket.get_key(url.path[1:])
+        key = key if key is not None else bucket.new_key(url.path[1:])
```

Comment:
Dubious about this line. If you left it at None, how much other code would have to change to accommodate that?

Also don't like `key = key if key …`. It's actually more code than a conditional assignment.

Alternatively, just do `key = bucket.get_key(…) or bucket.new_key(…)`